Userland/Libraries/LibWeb/Bindings/WindowObject.cpp

```diff
@@ -437,4 +449,167 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::inner_height_getter)
     return JS::Value(impl->inner_height());
 }
 
+JS_DEFINE_NATIVE_GETTER(WindowObject::scroll_x_getter)
+{
+    auto* impl = impl_from(vm, global_object);
+    if (!impl)
+        return {};
+    if (!impl->page())
+        return JS::Value(0.0);
+    return JS::Value(static_cast<double>(impl->page()->top_level_browsing_context().viewport_scroll_offset().x()));
```

Comment:
There's a `Value(i32)` ctor, which should fit here, no? `viewport_scroll_offset()` returns an `IntRect`.

Reply:
The spec defines it as returning a double, I didn't know if that was important or not? (Same with the 0.0s above.) If ints are fine then I'll do that. :+1:
> [Replaceable] readonly attribute double scrollX;

Reply:
This is basically an optimization - all numbers in JS are supposed to be doubles, but we have a JS::Value variant for numbers that are known to be integers in the i32 range - like this one. They'll automatically cast to double for operations that need it, but need only 32 bits for storage.

Reply:
Got it. Thanks!